pkg/apis/provisioning/v1alpha5/provisioner_validation.go

```diff
@@ -90,6 +90,9 @@ func (c *Constraints) validateLabels() (errs *apis.FieldError) {
 
 func IsRestrictedLabelDomain(key string) bool {
 	labelDomain := getLabelDomain(key)
+	if labelDomain == "kops.k8s.io" {
```

Comment:
Can you implement something similar to RestrictedLabelDomains (e.g. AllowedLabelDomains) and generalize this bit of code?

Reply:
That I can.

Reply:
Take a look if this is better now.